src/components/UserDetail.jsx

```diff
@@ -34,6 +39,34 @@ export default class UserDetail extends React.Component {
       console.log('loading comments for user ' + nextProps.selectedUser._id);
       nextProps.dispatch(fetchCommentsByUser(nextProps.selectedUser._id));
     }
+    this.updateTagsList(nextProps.selectedUser);
+  }
+
+  updateTagsList(user) {
+    this.tags = [];
+    if (user && user.tags && user.tags.length) {
+      for (var i in user.tags) {
+        this.tags.push({
+          id: user.tags[i] + Math.random(),
+          text: user.tags[i]
+        });
+      }
+    }
+  }
+
+  onTagsChange(tags) {
+    if (this.props.selectedUser && this.props.selectedUser._id) {
+      var preparedUser = {};
+      preparedUser['id'] = this.props.selectedUser._id;
+      preparedUser['name'] = this.props.selectedUser.user_name;
+      preparedUser['avatar'] = this.props.selectedUser.avatar;
+      preparedUser['status'] = this.props.selectedUser.status;
+      preparedUser['tags'] = [];
+      for(var i in tags) {
+        preparedUser['tags'].push(tags[i].text);
+      }
```

Comment:
A couple of things here.
- `for in` loops are generally used for iterating over Objects in javascript since some weirdness can happen with prototype chains. 
- since `for` loops with counters are even less intuitive when reading code, I generally use `map` when solving the problem above to give some hint to other programmers, like so:

`preparedUser.tags = tags.map(tag => tag.text);`